crates/consensus/src/transaction/envelope.rs

```diff
@@ -62,12 +62,17 @@ impl TryFrom<u8> for TxType {
 ///
 /// [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(tag = "type"))]
```

Comment:
I think that type may be omitted for legacy txns, and any objects generated in the past and stored somewhere will definitely have type omitted. So this would prevent us from handling old receipts at the very least

Reply:
@prestwich so those should be untagged instead?

Reply:
ideally we would write custom impls here I think? but I don't feel super strongly about it. can you check ethers-rs and its issues for how we used to do this? as long as we're not _worse_ than ethers, we should be fine to move forward

Reply:
I thin i'm comfortable leaving as typed, as supporting outdated nodes or long-term stored responses is not a compelling usecase. 

> imo ideal impl here would be the one which is able to deserialize any valid eth_getTransactionByHash response into TxEnvelope and per execution spec all RPC responses must include "type"

agree, tho as stated elsewhere, i think the "right" way to achieve this is by embedding `TxEnvelope` in the transaction rpc response type

Reply:
> agree, tho as stated elsewhere, i think the "right" way to achieve this is by embedding TxEnvelope in the transaction rpc response type

perhaps this is the next step once network abstraction is generic enough to allow us such strict constraints on RPC responses without breaking stuff for projects using `Ethereum` network for everything?

Reply:
the way to enforce it as a bound on network (if we even want to do that) would be to have `type TransactionResponse: RpcObject + AsRef<Self::TxEnvelope>`, basically. That would ensure that any response object contained an envelope

I think that the longer we go **without** embedding consensus objects in the RPC types, the worse of a time we'll have making the change

My primary target list is 
- Log
- TxEnvelope
- Receipt